xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/PropertyGroupDescriptor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.properties;
+
+import java.util.List;
+
+/**
+ * Contains a {@code List<String>} to view the group as a single object.
+ *
+ * @version $Id$
+ * @since 10.10RC1
+ */
+public class PropertyGroupDescriptor
+{
+    private List<String> group;
+
+    /**
+     * Default constructor.
+     *
+     * @param group hierarchy of groups
+     */
+    public PropertyGroupDescriptor(List<String> group)
+    {
+        this.group = group;
```

Comment:
Would be safer to do something like

`this.groups = Collections.unmodifiableList(group)`

otherwise anyone can modify the actual path trough getGroup() which is not great for a public API.